distributed/cli/dask_worker.py

```diff
@@ -295,23 +305,27 @@ def main(
         if v is not None
     }
 
-    if nprocs == "auto":
-        nprocs, nthreads = nprocesses_nthreads()
+    if nprocs is not None:
```

Comment:
Users could specify both `--nprocs` and `--num-workers` at the same time (though this is definitely an edge case). Currently we'll emit a warning and use the `--nprocs` value, which may be confusing. Instead we should raise an error in the case that both `--nprocs` and `--num-workers` is provided, since it's not clear which one should be used.